### PR TITLE
Adding doxygen documentation and READMEs to packages

### PR DESCRIPTION
**Pull request by brawner**
Completing the API documentation of rmw. I tried to research as much as I could through rcl and rcutils packages which this package borrows from as well as design.ros2.org documentation.

One question I had for the maintainers was whether items that were strictly defined to a corresponding element in rcutils should have documention, or just link to rcutils documentation? For those, I copied the existing comments, but that does create two locations where that documentation has to be maintained.

Signed-off-by: Stephen Brawner <brawner@gmail.com>

### CODE REVIEW

File: rmw/include/rmw/rmw.h [hunk 1]
```diff
@@ -824,6 +828,15 @@ rmw_return_loaned_message_from_subscription(
   const rmw_subscription_t * subscription,
   void * loaned_message);
 
+/// Create an rmw client to communicate with the specified service
+/**
+ * \param node Handle to node with which to register this client
+ * \param type_support The type_support of this rosidl service
+ * \param service_name The name of the ROS 2 service to connect with
+ * \param qos_policies The QoS profile policies to utilize for this connection
+ *
```
Comment (ahcorde) [inline rmw/include/rmw/rmw.h:837]: Remove line

File: rmw/include/rmw/convert_rcutils_ret_to_rmw_ret.h [hunk 2]
```diff
@@ -25,6 +25,11 @@ extern "C"
 #include "rmw/types.h"
 #include "rmw/visibility_control.h"
 
+/// Map a `rcutil_ret_t` value to the equivalent `rmw_ret_t` value
+/**
+ * \param rcutils_ret The rcutils type to map
+ * \param rmw_ret_t result mapped value, or RMW_RET_ERROR if the rcutils_ret value is not supported
+ */
```
[the remark this answers is not in the document]
Comment (brawner) [reply rmw/include/rmw/convert_rcutils_ret_to_rmw_ret.h:32]: Fixed

File: rmw/include/rmw/rmw.h (hunk 1 above)
Comment (brawner) [reply rmw/include/rmw/rmw.h]: Fixed

**Comment by brawner** [comment]
Thanks for the feedback @ahcorde. Should be addressed.

**Comment by brawner** [comment]
CI status:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10461/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5949/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8533/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/10342/)

File: rmw/include/rmw/topic_endpoint_info_array.h [hunk 3]
```diff
@@ -89,6 +93,8 @@ rmw_topic_endpoint_info_array_init_with_size(
  * This function reclaims any allocated resources within the object and also sets the value of size
  * to 0.
  *
+ * If a non RMW_RET_OK return value is returned, the RMW error message will be set
```
Comment (dirk-thomas) [inline rmw/include/rmw/topic_endpoint_info_array.h:96]: Trailing space introduced linter warning. Fixed in #224.

**Pull request merged**